### ISSUE TITLE
Pylint issues useless-super-delegation instead of useless-constructor

### ISSUE THREAD

**Issue by bgedik** [PyCQA/pylint#1567]
### Steps to reproduce
1. Create a file,  `test.py` that has the following code:

```python
"""Test"""

# pylint: disable=too-few-public-methods
class Base(object):

    """Base class"""

    def __init__(self):
        print Base.__name__

# pylint: disable=too-few-public-methods
class DerivedA(Base):

    """DerivedA class"""

    def __init__(self):
        super(DerivedA, self).__init__()

# pylint: disable=too-few-public-methods
class DerivedB(Base):

    """DerivedB class"""

    def __init__(self):
        Base.__init__(self)

# pylint: disable=too-few-public-methods
class DerivedC(Base):

    """DerivedC class"""

    pass
```
Run `pylint test.py`.

### Current behavior
Produces the following warning:
 `W: 16, 4: Useless super delegation in method '__init__' (useless-super-delegation)`

### Expected behavior
1) A warning should be given for both `DerivedA` and `DerivedB`
2) The problem is not about super delegation. It is about the useless constructor itself. The correct implementation is `DerivedC`.

### pylint --version output
No config file found, using default configuration
pylint 1.7.2, 
astroid 1.5.3
Python 2.7.12 (default, Nov 19 2016, 06:48:10) 
[GCC 5.4.0 20160609]

**Comment by PCManticore** [comment]
Thanks, this makes sense actually. useless-super-delegation is just useless-constructor in disguise. We could rename useless-super-delegation in a backwards compatible way and let it catch DerivedB as well.

**Comment by hippo91** [comment]
I'm not sure to understand. ``useless-super-delegation`` is in this *specific* case a useless-constructor in disguise but not always as shown by the functional test ``useless-super-delegation.py``.
We could check ``useless-super-delegation`` for ``__init__`` method and then raise a ``useless-constructor`` message.

**Comment by PCManticore** [comment]
Agree with @hippo91 This check targets the `super()` delegation while an `useless-constructor` would be a separate concern.

**Comment by Stevoisiak** [comment]
I agree that `useless-super-delegation` is a bit misleading in this case. If the entire constructor is useless, a `useless-constructor` warning should take priority.

**Comment by Stevoisiak** [comment]
It looks like the `super` issue is more complex than it seems. Removing the `super` delegation affects multi-class inheritance.

```Python
class Base1(object):
    def __init__(self):
        super(Base1, self).__init__()

class Base2(object):
    def __init__(self):
        object.__init__(self)

class Base3(object):
    def __init__(self):
        pass

class Base4(object):
    pass

class Second:
    def __init__(self):
        print("second")
        super().__init__()

class Derived1(Base1, Second):
    def __init__(self):
        print("Derived1 init")
        super().__init__()

class Derived2(Base2, Second):
    def __init__(self):
        print("Derived2 init")
        super().__init__()

class Derived3(Base3, Second):
    def __init__(self):
        print("Derived3 init")
        super().__init__()

class Derived4(Base4, Second):
    def __init__(self):
        print("Derived4 init")
        super().__init__()

Derived1()
Derived2()
Derived3()
Derived4()
```

Output

```
Derived1 init
second
Derived2 init
Derived3 init
Derived4 init
second
```
